akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala

```diff
@@ -392,6 +396,40 @@ private[projection] class SlickProjectionImpl[Offset, Envelope, P <: JdbcProfile
 
     private[projection] def newRunningInstance(): RunningProjection =
       new SlickRunningProjection(RunningProjection.withBackoff(() => mappedSource(), settings), this)
+
+    /**
+     * A convenience method to serialize asynchronous operations to occur one after another is complete
+     */
+    private def serialize(
+        batches: Map[String, Seq[(Offset, Envelope)]],
+        op: (String, Seq[(Offset, Envelope)]) => Future[Done],
+        logProgressEvery: Int = 5): Future[Done] = {
+      val size = batches.size
+      logger.info("Processing [{}] partitioned batches serially", size)
+
+      def loop(remaining: List[(String, Seq[(Offset, Envelope)])], n: Int): Future[Done] = {
+        remaining match {
+          case Nil => Future.successful(Done)
+          case (key, batch) :: tail =>
+            op(key, batch).flatMap { _ =>
+              if (n % logProgressEvery == 0)
+                logger.info("Processed batches [{}] of [{}]", n, size)
```

Comment:
I think we can remove the logging in this method. Not expecting very many batches, and it's already logging each one "Processing grouped envelopes"

Reply:
It depends on the user, but it's not out of the ordinary to have 10, 50, 100+ topic partitions.  The logging might be informational if users want to see what's causing delays.

I think this may be another clue that the abstraction isn't quite right.  If we had a projection per partition (i.e. the sub projections I mentioned earlier) then we wouldn't need to be concerned about the concurrency issues with the envelope handler